napari/layers/labels/labels.py

```diff
@@ -814,6 +821,7 @@ def show_selected_label(self, show_selected):
         self.colormap.selection = self.selected_label
         self.events.show_selected_label(show_selected_label=show_selected)
         self._cached_labels = None
+        self._cached_labels_mapping = {}
```

Comment:
Since these two are always together (?), I wonder whether we should have a method `_clear_caches`?

Reply:
done